streams/src/main/java/org/apache/kafka/streams/query/StateQueryResult.java

```diff
@@ -77,6 +79,12 @@ public QueryResult<R> getOnlyPartitionResult() {
                 "The query did not return exactly one partition result: " + partitionResults
             );
         } else {
+            if (nonempty.isEmpty() && partitionResults.size() != 0 && partitionResults.get(0).isFailure()) {
+                FailureReason failureReason = partitionResults.get(0).getFailureReason();
+                if (failureReason.equals(FailureReason.UNKNOWN_QUERY_TYPE)) {
+                    return QueryResult.forFailure(FailureReason.UNKNOWN_QUERY_TYPE, "unknown query type");
+                }
```

Comment:
I think we can use a for loop to count failure reason, if all unknown query type, we can return `QueryResult.forFailure(FailureReason.UNKNOWN_QUERY_TYPE, "unknown query type")` in the end.